Fix/reduce app size

baraa0abd opened the pull request:
New app size:

<img width="759" height="36" alt="image" src="https://github.com/user-attachments/assets/eda2e9f8-80c5-4237-8eab-66a6ff88564c" />


fav Icons : 
<img width="220" height="167" alt="image" src="https://github.com/user-attachments/assets/8ddf5cd3-e02e-49fa-a15f-3f1f13526c82" />

baraa0abd committed:
feat: Enable Proguard and R8 for release builds

- Enable code shrinking (minify) and resource shrinking for the Android release build to reduce app size.
- Add Proguard rules to ensure Flutter and its plugins work correctly with obfuscation.
- Add packaging options to exclude unnecessary metadata files, further reducing the APK size.
- Add `flutter_gap`, `flutter_launcher_icons`, and their transitive dependencies to `pubspec.lock`.

baraa0abd committed:
refactor: Standardize asset naming and update references

- Rename SVG assets to follow the `ic_` prefix convention for icons.
- Update `app_icons.dart` and `assets.dart` to reflect the new icon names.
- Remove unused image and SVG assets (`product_1.png`, `product_2.png`, `store_accessories.png`, `store_techno.png`, `sellio_splash.png`, `profile.svg`, `stars.svg`, `unselected-favourite.svg`).
- Update splash screen configuration in `pubspec.yaml` to use `sellio.png`.
- Replace placeholder images in various screens (`GridProduct`, `DataProvider`, `thrift_screen`, `products_section`) with existing assets.

baraa0abd committed:
Merge remote-tracking branch 'origin/develop' into fix/reduce-appSize

# Conflicts:
#	lib/core/design_system/constants/assets.dart

baraa0abd committed:
chore: Update dependencies and pubspec configuration

- Broadens the Dart SDK constraint to `>=3.5.0 <4.0.0`.
- Upgrades various dependencies, including `flutter_svg`, `url_launcher`, and `google_maps_flutter`, to their latest versions.
- Moves `flutter_native_splash` from `dependencies` to `dev_dependencies` to reflect its tooling nature.
- Adds comments to `pubspec.yaml` to categorize dependencies and clarify configuration sections for assets, fonts, and native splash screens.
- Adds `adaptive_icon_background` and `adaptive_icon_foreground` to the `flutter_launcher_icons` configuration for Android adaptive icons.

baraa0abd committed:
refactor: Replace PNG images with WEBP format to reduce app size

baraa0abd committed:
feat: Enable deferred components and optimize Android release build

- Enables deferred components by adding the Google Play Core libraries and enabling `multiDex`.
- Adds Proguard rules to support Play Core and Flutter's deferred component classes.
- Further optimizes the Android release build by disabling debug flags (`isDebuggable`, `isJniDebuggable`).
- Expands packaging options to exclude `META-INF/INDEX.LIST`.
- Updates a placeholder product image reference in `store_data_provider.dart`.

baraa0abd committed:
chore: Conditionally apply release signing configuration

- Updates `android/app/build.gradle.kts` to only apply the release signing configuration if the keystore properties file exists.
- This prevents build failures in environments where the signing key is not present, such as on CI/CD systems or for developers without access to the release keystore.

baraa0abd committed:
refactor: Simplify Android build configuration

- Removes redundant comments from `android/app/build.gradle.kts`.
- Simplifies the `buildTypes` and `packagingOptions` blocks by removing explanatory comments.

ghost commented:
<!-- sellio-bot-autoupdate-comment -->
**[Sellio Bot] Auto-update Status**


✅ Branch successfully updated from `develop`.

YaseenOmar reviewed (changes requested):
The favorite icon doesn’t seem to work correctly — please check its state and behavior.

<img width="375" height="173" alt="Screenshot from 2025-10-31 13-25-46" src="https://github.com/user-attachments/assets/7a0109d8-df31-4bfd-9695-1809f2f8b7eb" />

baraa0abd commented:
> The favorite icon doesn’t seem to work correctly — please check its state and behavior.
> 
> <img alt="Screenshot from 2025-10-31 13-25-46" width="375" height="173" src="https://private-user-images.githubusercontent.com/93636719/508151979-7a0109d8-df31-4bfd-9695-1809f2f8b7eb.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjE5MDgyMjksIm5iZiI6MTc2MTkwNzkyOSwicGF0aCI6Ii85MzYzNjcxOS81MDgxNTE5NzktN2EwMTA5ZDgtZGYzMS00YmZkLTk2OTUtMTgwOWYyZjhiN2ViLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEwMzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMDMxVDEwNTIwOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk3ODMxMGYyMzk0ZGU0YjQwMjhhZjNmZDMxMzc1ZGEzODlhMmVkMDliZmY1ZGY3MjI1ZjlhNjgxN2UzZWU2ZWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Aeqn5htNOLC8glgaw2zE4ADaKOBQuKv8haS3Hi5imr8">

Actually I didnot change anything with the icons , just rename it to be like ic_icon_name (and delete unused once)

baraa0abd committed:
refactor: Replace favorite icon and update asset paths

- Deletes the old `ic_favourite_icon.svg`.
- Adds a new, smaller `ic_unselected-favourite.svg` icon.
- Updates references in `assets.dart` to point to the new icon file for both `favouriteIcon` and `unselectedFavorite` constants.

ghost commented:
<!-- sellio-bot-autoupdate-comment -->
**[Sellio Bot] Auto-update Status**


✅ Branch successfully updated from `develop`.

YaseenOmar reviewed (dismissed):
Great job , Baraa

mohamedebrahem13 committed:
coverts missing images

ghost commented:
<!-- sellio-bot-autoupdate-comment -->
**[Sellio Bot] Auto-update Status**


✅ Branch successfully updated from `develop`.

ghost commented:
<!-- sellio-bot-autoupdate-comment -->
**[Sellio Bot] Auto-update Status**


✅ Branch successfully updated from `develop`.
